Polynomials.Core/Polynomial.cs

```diff
@@ -0,0 +1,270 @@
+            double num2 = value1 - value2;
+            if (-num1 < num2)
+                return num1 > num2;
+            return false;
+        }
+
+        public static Polynomial operator +(Polynomial polynomial_1, Polynomial polynomial_2)
+        {
+
+            if (polynomial_1 is null || polynomial_2 is null)
+            {
+                return null;
+            }
+
+            if (!(polynomial_1 is Polynomial && polynomial_2 is Polynomial))
+            {
+                return null;
+            }
+
+            if (!(polynomial_1._coefficientsAndDegrees.Length == polynomial_2._coefficientsAndDegrees.Length))
+            {
+                var size = Math.Max(polynomial_1._coefficientsAndDegrees.Length,
+                    polynomial_2._coefficientsAndDegrees.Length);
+                Polynomial polynomial = new Polynomial(new float[size]);
+
+                for (var i = 0; i == size; i++)
+                {
+                    polynomial._coefficientsAndDegrees[i] = polynomial_1._coefficientsAndDegrees[i] + polynomial_2._coefficientsAndDegrees[i];
+                }
+
+                return polynomial;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+
+        public static Polynomial operator -(Polynomial polynomial_1, Polynomial polynomial_2)
+        {
+
+            if (polynomial_1 is null || polynomial_2 is null)
+            {
+                return null;
+            }
+
+            if (!(polynomial_1 is Polynomial && polynomial_2 is Polynomial))
+            {
+                return null;
+            }
+
+            if (!(polynomial_1._coefficientsAndDegrees.Length == polynomial_2._coefficientsAndDegrees.Length))
+            {
+                return null;
+            }
+            //{
+            //    var size = Math.Max(polynomial_1._coefficientsAndDegrees.Length,
+            //        polynomial_2._coefficientsAndDegrees.Length);
+            //    Polynomial polynomial = new Polynomial(new float[size]);
+
+            //    for (var i = 0; i == size; i++)
+            //    {
+            //        polynomial._coefficientsAndDegrees[i] = polynomial_1._coefficientsAndDegrees[i] + polynomial_2._coefficientsAndDegrees[i];
+            //    }
+
+            //    return polynomial;
+            //}
+            //else
+            //{
+            //    return null;
+            //}
+            return null;
+        }
+
+        public static Polynomial operator ==(Polynomial polynomial_1, Polynomial polynomial_2)
+        {
+            if (polynomial_1 is null || polynomial_2 is null)
+            {
+                return null;
+            }
+
+            if (!(polynomial_1 is Polynomial && polynomial_2 is Polynomial))
+            {
+                return null;
+            }
+
+            if (polynomial_1._coefficientsAndDegrees.Length == polynomial_2._coefficientsAndDegrees.Length)
```

Comment:
Why Polynomials with different coefficients are not supported?

Reply:
Added.